.github/workflows/semantic_library_workflow.yml

```
@@ -121,6 +121,13 @@ jobs:
           java_version: ${{ inputs.java_version }}
           setup_android_sdk: 'false'
 
+  validation:
+    name: "Validation"
```

Can we call this "Wrapper Validation" to better indicate what is being validated? This will be useful when viewing workflow runs too.
```suggestion
  wrapper_validation:
    name: "Wrapper Validation"
```

Also there are other jobs that may use the gradle wrapper such as `code_lint` and `instrumentation_tests`. Since the wrapper validation action is fast and `code_lint` looks like a parent of all gradle related tasks maybe we can just make `code_lint` need `wrapper_validation` and remove that requirement from the other jobs?

---

Cool, thanks, Here is the process:
1, according to spec [here](https://github.com/gradle/wrapper-validation-action/tree/releases/v2?tab=readme-ov-file), this step should precede any step using gradle/gradle-build-action or gradle/actions/setup-gradle.

2, in Shared-CI-Action-Android, unit-test, build, code coverage, publish and gradle task contain gradle/gradle-build-action. Considering code_lint will execute gradle/gradle-build-action in an if statement, and build, unit-test both depend on code lint, validation is added into code lint now.